homeassistant/components/freebox/binary_sensor.py

```diff
@@ -47,6 +53,111 @@ async def async_setup_entry(
     if binary_entities:
         async_add_entities(binary_entities, True)
 
+    for nodeid, node in router.home_devices.items():
+        if nodeid in tracked:
+            continue
+        if node["category"] == "pir":
+            new_trackedpir.append(FreeboxPir(hass, router, node))
+        elif node["category"] == "dws":
+            new_trackeddws.append(FreeboxDws(hass, router, node))
+
+        sensor_cover_node = next(
+            filter(
+                lambda x: (x["name"] == "cover" and x["ep_type"] == "signal"),
+                node["show_endpoints"],
+            ),
+            None,
+        )
+        if sensor_cover_node and sensor_cover_node.get("value") is not None:
+            new_trackedcover.append(FreeboxSensorCover(hass, router, node))
+
+        tracked.add(nodeid)
+
+    if new_trackedpir:
+        async_add_entities(new_trackedpir, True)
+    if new_trackeddws:
+        async_add_entities(new_trackeddws, True)
+    if new_trackedcover:
+        async_add_entities(new_trackedcover, True)
+
+
+class FreeboxPir(FreeboxHomeEntity, BinarySensorEntity):
+    """Representation of a Freebox motion binary sensor."""
+
+    def __init__(
+        self, hass: HomeAssistant, router: FreeboxRouter, node: dict[str, Any]
+    ) -> None:
+        """Initialize a Pir."""
+        super().__init__(hass, router, node)
+        self._command_trigger = self.get_command_id(
+            node["type"]["endpoints"], "signal", "trigger"
+        )
+
+        self._detection = False
+
+    async def async_update_signal(self):
+        """Watch states."""
+        detection = await self.get_home_endpoint_value(self._command_trigger)
+        if detection is not None:
+            if self._detection == detection:
+                self._detection = not detection
```

Comment:
The api returns a value when the start/stop detection is triggered (detection ON and detection OFF)

Reply:
Okeyyyy

Having an API return to None/null is possible ?

Otherwise `self._detection = not await self.get_home_endpoint_value(self._command_trigger)`